endgame_simulations/simulations.py

```diff
@@ -0,0 +1,232 @@
+from abc import ABC, abstractproperty
+from typing import ClassVar, Generic, Iterator, TypeVar, cast, overload
+
+import tqdm
+from hdf5_dataclass import FileType
+
+from endgame_simulations.models import BaseInitialParams
+
+from .common import AdvanceState, BaseState, State
+
+ParamsModel = TypeVar("ParamsModel", bound=BaseInitialParams)
+
+
+class GenericSimulation(Generic[ParamsModel, State], ABC):
```

Comment:
Hmm, this is remarkably similar to EndgameSimulation. As far as I can see the differences are: 1) reset_current_params only in GenericSimulation 2) EndgameSimulation's init does one operations extra (think super().__init__ followed by that operation)

Reply:
Yeah I agree, as it is currently. Still need to finish writing endgame sim, this PR isn't really for that (it's really just come along for the ride!) When I have the test working, I would suggest we also implement in epioncho - at that point we can talk about AbstractCommon